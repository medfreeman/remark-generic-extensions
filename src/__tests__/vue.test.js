import { transformToVue } from "./helpers/transform";

describe("vue inline extensions", () => {
  test("should add an extension", async () =>
    expect(await transformToVue("!Icon")).toMatchSnapshot());

  test("should add an extension with an id", async () =>
    expect(await transformToVue("!Icon{ #my-id }")).toMatchSnapshot());

  test("should add an extension with a class", async () =>
    expect(await transformToVue("!Icon{ .my-class }")).toMatchSnapshot());

  test("should add an extension with an attribute", async () =>
    expect(await transformToVue("!Icon{ attr=my-attr }")).toMatchSnapshot());

  test("should add an extension with a quoted attribute", async () =>
    expect(
      await transformToVue(`!Icon{ attr="my attribute" }`)
    ).toMatchSnapshot());

  test("should add an extension with a lone attribute", async () =>
    expect(await transformToVue(`!Icon{ attr }`)).toMatchSnapshot());

  test("should add an extension with a lone attribute with a default value", async () =>
    expect(
      await transformToVue(`!Icon{ attr }`, {
        elements: {
          Icon: {
            propsDefaultValues: {
              attr: "default"
            }
          }
        }
      })
    ).toMatchSnapshot());

  test("should add an extension with a lone attribute with a default boolean value", async () =>
    expect(
      await transformToVue(`!Icon{ highlight }`, {
        elements: {
          Icon: {
            propsDefaultValues: {
              highlight: true
            }
          }
        }
      })
    ).toMatchSnapshot());

  test("should add an extension with all types of parameters", async () =>
    expect(
      await transformToVue(
        `!Icon{ #my-id .my-class attr attr2=my-attr attr3="my attribute" }`,
        {
          elements: {
            Icon: {
              propsDefaultValues: {
                attr: "default"
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should add an extension with a boolean property", async () =>
    expect(
      await transformToVue("!Icon", {
        elements: {
          Icon: {
            html: {
              properties: {
                highlight: true
              }
            }
          }
        }
      })
    ).toMatchSnapshot());

  test("should add an extension with a mapped content", async () =>
    expect(
      await transformToVue("!Icon[my-tooltip]", {
        elements: {
          Icon: {
            html: {
              properties: {
                tooltip: "::content::"
              }
            }
          }
        }
      })
    ).toMatchSnapshot());

  test("should add an extension with a mapped argument", async () =>
    expect(
      await transformToVue("!Icon(my-icon)", {
        elements: {
          Icon: {
            html: {
              properties: {
                icon: "::argument::"
              }
            }
          }
        }
      })
    ).toMatchSnapshot());

  test("should work with other markdown", async () =>
    expect(
      await transformToVue(
        `# heading

!Icon`
      )
    ).toMatchSnapshot());
});

describe("vue block extensions", () => {
  test("should add an extension", async () =>
    expect(
      await transformToVue(
        `Icon:
:::
:::`
      )
    ).toMatchSnapshot());

  test("should add an extension with an id", async () =>
    expect(
      await transformToVue(
        `Icon:
:::
:::
{ #my-id }`
      )
    ).toMatchSnapshot());

  test("should add an extension with a class", async () =>
    expect(
      await transformToVue(
        `Icon:
:::
:::
{ .my-class }`
      )
    ).toMatchSnapshot());

  test("should add an extension with an attribute", async () =>
    expect(
      await transformToVue(
        `Icon:
:::
:::
{ attr=my-attr }`
      )
    ).toMatchSnapshot());

  test("should add an extension with a quoted attribute", async () =>
    expect(
      await transformToVue(
        `Icon:
:::
:::
{ attr="my attribute" }`
      )
    ).toMatchSnapshot());

  test("should add an extension with a lone attribute", async () =>
    expect(
      await transformToVue(
        `Icon:
:::
:::
{ attr }`
      )
    ).toMatchSnapshot());

  test("should add an extension with a lone attribute with a default value", async () =>
    expect(
      await transformToVue(
        `Icon:
:::
:::
{ attr }`,
        {
          elements: {
            Icon: {
              propsDefaultValues: {
                attr: "default"
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should add an extension with a lone attribute with a default boolean value", async () =>
    expect(
      await transformToVue(
        `Icon:
:::
:::
{ highlight }`,
        {
          elements: {
            Icon: {
              propsDefaultValues: {
                highlight: true
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should add an extension with all types of parameters", async () =>
    expect(
      await transformToVue(
        `Icon:
:::
:::
{ #my-id .my-class attr attr2=my-attr attr3="my attribute" }`,
        {
          elements: {
            Icon: {
              propsDefaultValues: {
                attr: "default"
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should add an extension with a boolean property", async () =>
    expect(
      await transformToVue(
        `Icon:
:::
:::`,
        {
          elements: {
            Icon: {
              html: {
                properties: {
                  highlight: true
                }
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should add an extension with a mapped content", async () =>
    expect(
      await transformToVue(
        `Icon:
:::
my-tooltip
:::`,
        {
          elements: {
            Icon: {
              html: {
                properties: {
                  tooltip: "::content::"
                }
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should add an extension with a mapped argument", async () =>
    expect(
      await transformToVue(
        `Icon: my-icon
:::
:::`,
        {
          elements: {
            Icon: {
              html: {
                properties: {
                  icon: "::argument::"
                }
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should work with other markdown", async () =>
    expect(
      await transformToVue(
        `# heading

Icon:
:::
:::`
      )
    ).toMatchSnapshot());
});
