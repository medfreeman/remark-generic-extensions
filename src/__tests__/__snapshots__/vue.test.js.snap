// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue block extensions should add an extension 1`] = `
<div>
  <icon></icon>
</div>
`;

exports[`vue block extensions should add an extension with a boolean property 1`] = `
<div>
  <icon highlight="true"></icon>
</div>
`;

exports[`vue block extensions should add an extension with a class 1`] = `
<div>
  <icon class="my-class"></icon>
</div>
`;

exports[`vue block extensions should add an extension with a lone attribute 1`] = `
<div>
  <icon attr=""></icon>
</div>
`;

exports[`vue block extensions should add an extension with a lone attribute with a default boolean value 1`] = `
<div>
  <icon highlight="true"></icon>
</div>
`;

exports[`vue block extensions should add an extension with a lone attribute with a default value 1`] = `
<div>
  <icon attr="default"></icon>
</div>
`;

exports[`vue block extensions should add an extension with a mapped argument 1`] = `
<div>
  <icon icon="my-icon"></icon>
</div>
`;

exports[`vue block extensions should add an extension with a mapped content 1`] = `
<div>
  <icon tooltip="my-tooltip"></icon>
</div>
`;

exports[`vue block extensions should add an extension with a quoted attribute 1`] = `
<div>
  <icon attr="my attribute"></icon>
</div>
`;

exports[`vue block extensions should add an extension with all types of parameters 1`] = `
<div>
  <icon attr3="my attribute" attr2="my-attr" class="my-class" id="my-id" attr="default"></icon>
</div>
`;

exports[`vue block extensions should add an extension with an attribute 1`] = `
<div>
  <icon attr="my-attr"></icon>
</div>
`;

exports[`vue block extensions should add an extension with an id 1`] = `
<div>
  <icon id="my-id"></icon>
</div>
`;

exports[`vue block extensions should work with other markdown 1`] = `
<div>
  <h1>heading</h1>
  <icon></icon>
</div>
`;

exports[`vue inline extensions should add an extension 1`] = `
<div>
  <p>
    <icon></icon>
  </p>
</div>
`;

exports[`vue inline extensions should add an extension with a boolean property 1`] = `
<div>
  <p>
    <icon highlight="true"></icon>
  </p>
</div>
`;

exports[`vue inline extensions should add an extension with a class 1`] = `
<div>
  <p>
    <icon class="my-class"></icon>
  </p>
</div>
`;

exports[`vue inline extensions should add an extension with a lone attribute 1`] = `
<div>
  <p>
    <icon attr=""></icon>
  </p>
</div>
`;

exports[`vue inline extensions should add an extension with a lone attribute with a default boolean value 1`] = `
<div>
  <p>
    <icon highlight="true"></icon>
  </p>
</div>
`;

exports[`vue inline extensions should add an extension with a lone attribute with a default value 1`] = `
<div>
  <p>
    <icon attr="default"></icon>
  </p>
</div>
`;

exports[`vue inline extensions should add an extension with a mapped argument 1`] = `
<div>
  <p>
    <icon icon="my-icon"></icon>
  </p>
</div>
`;

exports[`vue inline extensions should add an extension with a mapped content 1`] = `
<div>
  <p>
    <icon tooltip="my-tooltip"></icon>
  </p>
</div>
`;

exports[`vue inline extensions should add an extension with a quoted attribute 1`] = `
<div>
  <p>
    <icon attr="my attribute"></icon>
  </p>
</div>
`;

exports[`vue inline extensions should add an extension with all types of parameters 1`] = `
<div>
  <p>
    <icon attr3="my attribute" attr2="my-attr" class="my-class" id="my-id" attr="default"></icon>
  </p>
</div>
`;

exports[`vue inline extensions should add an extension with an attribute 1`] = `
<div>
  <p>
    <icon attr="my-attr"></icon>
  </p>
</div>
`;

exports[`vue inline extensions should add an extension with an id 1`] = `
<div>
  <p>
    <icon id="my-id"></icon>
  </p>
</div>
`;

exports[`vue inline extensions should work with other markdown 1`] = `
<div>
  <h1>heading</h1>
  <p>
    <icon></icon>
  </p>
</div>
`;
